File: leetcode/find_pivot_index/find_pivot_index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        vector<int> left_sums;
        left_sums.push_back(0);
        for(int i=0;i<nums.size();i++){
            int prev_left_sum=left_sums[i];
            int left_sum=prev_left_sum+nums[i];
            left_sums.push_back(left_sum);
        }

        int total_sum=left_sums.back();

        for(int i=0;i<nums.size();i++){
            if((total_sum-nums[i])==(2*left_sums[i])){
                return i;
            }
        }
        return -1;
    }
};

File: leetcode/find_pivot_index/find_pivot_index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        prefix_sum={}

        total_sum=0
        for idx, num in enumerate(nums):
            total_sum+=num
            prefix_sum[idx]=total_sum

        for idx, num in enumerate(nums):
            left_sum=prefix_sum[idx]-num
            right_sum=total_sum-prefix_sum[idx]

            if left_sum==right_sum:
                return idx
        
        return -1